Add restricted method text (#1217)
MetaMask/metamask-docs: snaps/how-to/request-permissions.md

Snaps and dapps follow the [EIP-2255 wallet permissions specification](https://eips.ethereum.org/EIPS/eip-2255).

  :::info note

>   The [JSON-RPC](/wallet/reference/json-rpc-api) methods [`wallet_snap`](/wallet/reference/wallet_snap) and [`wallet_invokeSnap`](/wallet/reference/wallet_invokesnap) are restricted. To use these restricted methods, you must first obtain permission through [`wallet_requestSnap`](/wallet/reference/wallet_requestSnaps).

Reviewer: Should be [`wallet_requestSnaps`]

  :::

> 

Reviewer 2: This is not needed in this location. There is a "Request permissions from a dapp" section at the bottom of this page that mentions `wallet_requestSnaps`. You can add mention of `wallet_snap` and `wallet_invokeSnap` in that section.

> Dapps that communicate with Snaps must request permission to do so by calling the

Reviewer 2: ```suggestion
Dapps that communicate with Snaps using [`wallet_snap`](/wallet/reference/wallet_snap) and [`wallet_invokeSnap`](/wallet/reference/wallet_invokesnap) must request permission to do so by calling the
```

[`wallet_requestSnaps`](/wallet/reference/wallet_requestsnaps) MetaMask JSON-RPC API method.

> :::info note
> Restricted [JSON-RPC](/wallet/reference/json-rpc-api) methods [`wallet_snap`](/wallet/reference/wallet_snap) 
> and [`wallet_invokeSnap`](/wallet/reference/wallet_invokesnap) require obtaining permission 
> through [`wallet_requestSnaps`](/wallet/reference/wallet_requestSnaps) before use.
> :::
> 

Reviewer 2: ```suggestion
```

> For example, to request permission to connect to the `hello-snap` Snap:

Reviewer 2: ```suggestion
The following example calls `wallet_requestSnaps` to request permission to connect to the `hello-snap` Snap, then calls `wallet_invokeSnap` to invoke the `hello` JSON-RPC method exposed by the Snap:
```

```js title="index.js"
// If the Snap is not already installed, the user will be prompted to install it.
await window.ethereum.request({
  method: 'wallet_requestSnaps',
  params: {
    // Assuming the Snap is published to npm using the package name 'hello-snap'.
    'npm:hello-snap': {},
  },
});

// Invoke the 'hello' JSON-RPC method exposed by the Snap.
const response = await window.ethereum.request({
  method: 'wallet_invokeSnap',
  params: { snapId: 'npm:hello-snap', request: { method: 'hello' } },
});

console.log(response); // 'world!'
```

> See [About the Snaps APIs](/snaps/learn/about-snaps/apis/#custom-json-rpc-apis) for more information on implementing custom JSON-RPC APIs to communicate with dapps and other Snaps.

Reviewer 2: ```suggestion
:::note
Learn more about implementing [custom JSON-RPC APIs](../learn/about-snaps/apis.md#custom-json-rpc-apis) in a Snap.
:::
```
